spec/browser/middleware/with-graphql-query.spec.js

```diff
@@ -0,0 +1,54 @@
+import forge, { configs } from 'src/index'
+import WithGraphqlQuery from 'src/middlewares/with-graphql-query'
+import MockGateway from 'src/gateway/mock'
+import { mockRequest } from 'src/test'
+
+describe('Middleware / WithGraphqlQuery', () => {
+  const query = 'query MyQuery { someResource { totalCount }'
+  const variables = { variables: { ids: [1, 5, 11] } }
+
+  let client, originalConfig
+
+  afterEach(() => {
+    configs.gateway = originalConfig
+  })
+
+  beforeEach(() => {
+    if (!originalConfig) {
+      originalConfig = configs.gateway
+    }
+
+    configs.gateway = MockGateway
+    const manifest = {
+      host: 'http://batman',
+      middleware: [WithGraphqlQuery(query)],
```

Comment:
Another thing, like I said, this is a bit clumsy. Need to look into how to make it work for resource.